sdp/processors/langs/arabic.py

```diff
@@ -0,0 +1,197 @@
+        self.remove_punctuation = remove_punctuation
+        self.remove_tatweel = remove_tatweel
+        self.normalize_ligature = normalize_ligature
+        self.normalize = normalize
+        self.apply_canonical_decomposition = apply_canonical_decomposition
+        self.apply_canonical_decomposition_canonical_composition = (
+            apply_canonical_decomposition_canonical_composition
+        )
+        self.apply_compatability_decomposition_canonical_composition = apply_compatability_decomposition_canonical_composition
+
+    def process_dataset_entry(self, data_entry):
+        data_entry[self.output_text_key] = self.clean_data(
+            data_entry[self.input_text_key]
+        )
+        return [DataEntry(data=data_entry)]
+
+    def _remove_diacritics(self, text):
+        for char in DIACRITICS:
+            text = text.replace(char, '')
+        return text
+
+    def _remove_punctuation(self, text):
+        for char in PUNCTUATION_MARKS:
+            text = text.replace(char, '')
+        return text
+
+    def _normalize_teh(self, text):
+        text = text.replace(TEH_MARBUTA, HEH)
+        return text
+    
+    def _normalize_ligature(self, text):
+        LIGUATURES_PATTERN = re.compile(u"[" + u"".join(LIGATURES) + u"]", re.UNICODE)
+        return LIGUATURES_PATTERN.sub(u'%s%s' % (LAM, ALEF), text)
+    
+    def _normalize_alef(self, text):
+        ALEFS_PATTERN = re.compile(u"[" + u"".join(ALEFS) + u"]", re.UNICODE)
+        return re.sub(ALEFS_PATTERN, ALEF, text)
+
+    def _remove_extra_spaces(self, text):
+        text = re.sub(" +", " ", text)
+        return text
+
+    def _remove_empty_lines(self, text):
+        lines = text.split("\n")
+        return ("\n").join([line for line in lines if len(line) >= 1])
+
+    def _normalize(self, text):
+        text = self._remove_diacritics(text)
+        text = self._normalize_alef(text)
+        text = self._normalize_ligature(text)
+        text = self._normalize_teh(text)
+
+        return text
+
+    def clean_data(self, text):
+        if self.remove_extra_spaces:
+            text = self._remove_extra_spaces(text)
+        if self.remove_empty_lines:
+            text = self._remove_empty_lines(text)
+        if self.remove_diacritics:
+            text = self._remove_diacritics(text)
+        if self.remove_tatweel:
+            text = text.replace("ـ", "")
+        if self.remove_punctuation:
+            text = self._remove_punctuation(text)
+        if self.normalize_ligature:
+            text = self._normalize_ligature(text)
+        if self.normalize:
+            text = self._normalize(text)
+        if self.apply_canonical_decomposition:
+            text = unicodedata.normalize("NFD", text)
+        if self.apply_canonical_decomposition_canonical_composition:
+            text = unicodedata.normalize("NFC", text)
+        if self.apply_compatability_decomposition_canonical_composition:
+            text = unicodedata.normalize("NFKC", text)
```

Comment:
Do we need to add any check to make sure that if one of these is True, then other's are not True, or is any combination OK?